main.tf

```diff
@@ -1,174 +1,159 @@
-          access_scope = {
-            type = "cluster"
-          }
-        }
-      }
-    }
-    # https://docs.aws.amazon.com/eks/latest/userguide/access-policies.html#access-policy-permissions
-    # TODO: Additional roles that need to be created:
-    # AmazonEKSAdminViewPolicy?
-    # AmazonEKSEditPolicy
-    # AmazonEKSViewPolicy
-
-  }
-  tags = var.tags
-}
-
+resource "aws_iam_role" "admin_role" {
+  name = "eks_admin_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::766808016710:root" # Replace YOUR_AWS_ACCOUNT_ID with your actual AWS account ID
+        }
+        Action = "sts:AssumeRole"
+      },
+    ]
+  })
+
+  tags = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "admin_policy" {
+  role       = aws_iam_role.admin_role.name
+  policy_arn = "arn:aws:iam::aws:policy/PowerUserAccess"
+}
+
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "spacelift-created-vpc"
+  cidr = var.cidr
+
+  azs             = var.azs
+  private_subnets = var.private_subnet_cidrs
+  public_subnets  = var.public_subnet_cidrs
+
+  private_subnet_tags = {
+    Name = "private"
+  }
+
+  # When removing the Internet gateway it might have allocated from elastic IP addresses
+  # Turn off the nat_gateway to force the IP addresses to be removed
+  # > "Network vpc-0f30cfca319ebc521 has some mapped public address(es). Please unmap those public address(es) before detaching the gateway.""
+  create_igw         = true
+  enable_nat_gateway = true
+  enable_vpn_gateway = false
+  single_nat_gateway = true
+
+  # manage_default_security_group = true
+  # default_security_group_egress = []
+  # default_security_group_ingress = []
+
+  tags = merge(
+    var.tags,
+    {
+      Terraform   = "true"
+      Environment = "dev"
+    }
+  )
+}
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 20.12"
+  # version = "~> 20.9"
+
+  depends_on = [module.vpc]
+
+  cluster_name    = var.cluster_name
+  cluster_version = var.cluster_version
+
+  cluster_endpoint_public_access = true
+
+  cluster_addons = {
+    coredns = {
+      most_recent = true
+    }
+    kube-proxy = {
+      most_recent = true
+    }
+    vpc-cni = {
+      most_recent = true
+    }
+    aws-ebs-csi-driver = {
+      most_recent = true
+    }
+  }
+
+  vpc_id                    = module.vpc.vpc_id
+  subnet_ids                = module.vpc.private_subnets
+  control_plane_subnet_ids  = module.vpc.intra_subnets
+  cluster_security_group_id = module.vpc.default_security_group_id
+
+
+  # EKS Managed Node Group(s)
+  eks_managed_node_group_defaults = {
+    instance_types = ["m6i.large", "m5.large", "m5n.large", "m5zn.large"]
+  }
+
+  eks_managed_node_groups = {
+    one = {
+      name         = var.eks_nodeGroup
+      desired_size = 1
+      min_size     = 0
+      max_size     = 2

```

Comment:
This is still something I'd like to figure out, spot claimed on the call that we didn't need a managed node group.

Reply:
So the problem is that in order for the add ons to be installed in the first place EKS has to have at least 1 node running as that is where the tools are installed to. This means that in order for the EKS cluster to be "green" it has to have at least a single node.

After this point we can install spot ocean.

Even their docs here: https://github.com/spotinst/terraform-spotinst-ocean-eks?tab=readme-ov-file#following-modules-should-be-used-instead--

Specify that we should:

1. Use the AWS EKS provider to create an EKS cluster and it's resources
2. Then install the ocean controllers/k8s